examples/combobox/js/combobox-autocomplete.js

```diff
@@ -0,0 +1,558 @@
+  }
+
+  if (force || (!this.comboboxHasVisualFocus && !this.listboxHasVisualFocus && !this.hasHover)) {
+    this.setCurrentOptionStyle(false);
+    this.listboxNode.style.display = 'none';
+    this.comboboxNode.setAttribute('aria-expanded', 'false');
+    this.buttonNode.classList.remove('open');
+    this.setActiveDescendant(false);
+  }
+};
+
+/* combobox Events */
+
+ComboboxAutocomplete.prototype.handleComboboxKeyDown = function (event) {
+  var flag = false,
+    char = event.key,
+    altKey   = event.altKey;
+
+  if (event.ctrlKey || event.shiftKey) {
+    return;
+  }
+
+  switch (event.key) {
+
+    case "Enter":
+      if (this.listboxHasVisualFocus) {
+        this.setValue(this.option.textContent);
+      }
+      this.close(true);
+      this.setVisualFocusCombobox();
+      flag = true;
+      break;
+
+    case "Down":
+    case "ArrowDown":
+      if (this.filteredOptions.length > 0) {
+        if (altKey) {
+          this.open();
+        }
+        else {
+          this.open();
+          if (this.listboxHasVisualFocus || (this.isBoth && this.filteredOptions.length > 1)) {
+            this.setOption(this.getNextOption(this.option), true);
+            this.setVisualFocusListbox();
+          }
+          else {
+            this.setOption(this.firstOption, true);
+            this.setVisualFocusListbox();
+          }
+        }
+      }
+      flag = true;
+      break;
+
+    case "Up":
+    case "ArrowUp":
+
+      if (this.hasOptions()) {
+        if (this.listboxHasVisualFocus) {
+          this.setOption(this.getPreviousOption(this.option), true);
+        }
+        else {
+          this.open();
+          if (!altKey) {
+            this.setOption(this.lastOption, true);
+            this.setVisualFocusListbox();
+          }
+        }
+      }
+      flag = true;
+      break;
+
+    case "Esc":
+    case "Escape":
+      if (this.isOpen()) {
+        this.close(true);
+        this.filter = this.comboboxNode.value;
+        this.filterOptions();
+        this.setVisualFocusCombobox();
+      }
+      else {
+        this.setValue('');
+        this.comboboxNode.value = '';
+      }
+      this.option = false;
+      flag = true;
+      break;
+
+    case "Tab":
+      this.close(true);
+      if (this.listboxHasVisualFocus) {
+        if (this.option) {
+          this.setValue(this.option.textContent);
+        }
+      }
+      break;
+
+    case "Home":
+      this.comboboxNode.setSelectionRange(0,0);
+      flag = true;
+      break;
+
+    case "End":
+      var length = this.comboboxNode.value.length;
+      this.comboboxNode.setSelectionRange(length,length);
+      flag = true;
+      break;
+
+
+    default:
+      break;
+  }
+
+  if (flag) {
+    event.stopPropagation();
+    event.preventDefault();
+  }
+
+};
+
+ComboboxAutocomplete.prototype.handleComboboxKeyUp = function (event) {
+  var flag = false,
+    option = false,
+    char = event.key;
+
+  function isPrintableCharacter (str) {
```

Comment:
it might make sense to define this outside of the `keyup` function

Reply:
Fixed in 2a5719e65bbd45bb8ae6ad05ac663051b8ed425f